optuna/testing/storages.py

```diff
@@ -71,7 +71,7 @@ def __enter__(
                 raise ValueError("InMemoryStorage does not accept any arguments!")
             return optuna.storages.InMemoryStorage()
         elif "sqlite" in self.storage_specifier:
-            self.tempfile = tempfile.NamedTemporaryFile()
+            self.tempfile = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
Is it possible to somehow delete the temporary files after the test? When you run the test in a local environment, temporary will survive until the machine is rebooted.

Reply:
Maybe it is possible, in this situation the problem is deleting tempfile **before** storage deletion (more specifically `storage.engine.dispose()`). Thus, calling `storage.enigne.dispose()` on the `__exit__` function will solve this problem.

Reply:
It's not that simple, it seems we have to use `os.unlink` to explicitly delete temporary files.

Reply:
This behavior is not a good thing. But I have no idea how to fix this. (I suspect many changes would be needed to fix this problem.)